### PR TITLE
Add guzzle promise onRejected handler

### LINKED ISSUE #17: Promise interface, onRejected ?
In `GuzzleRequestSpanning`, isn't the `$onRejected` callback missing ?

https://github.com/auxmoney/OpentracingBundle-Guzzle/blob/ced155c3335adf4341e22365470f5aa302dc16ef/Middleware/GuzzleRequestSpanning.php#L31-L37

See the relevant interface here:
https://github.com/guzzle/promises/blob/89b1a76b7fda5853401297dc4b2a093cba1fda23/src/PromiseInterface.php#L28-L31

I fear that if the request is canceled or in case of an error, the open span won't be closed.

### PR DESCRIPTION
**Pull request by eddycharly**
This PR fixes #17 

It adds a `$onRejected` handler on guzzle promises.
This ensures the started spans are closed even when a promise is rejected.

### CODE REVIEW

File: Middleware/GuzzleRequestSpanning.php [hunk 1]
```diff
@@ -33,6 +35,23 @@ function (ResponseInterface $response) {
...
+                            'error.object' => get_class($e),
+                            'message' => $e->getMessage(),
+                            'stack' => $e->getTraceAsString(),
+                        ]
+                    );
+                    if ($e->hasResponse()) {
+                        $this->requestSpanning->finish($e->getResponse()->getStatusCode());
+                    }
+                    $this->tracing->finishActiveSpan();
+                    throw $e;
```
Comment (cawolf) [inline Middleware/GuzzleRequestSpanning.php]: codeclimate:
> Method __invoke has 28 lines of code (exceeds 25 allowed). Consider refactoring. 

Maybe it is time to separate the callables into own methods within this class.

**Review by cawolf — CHANGES_REQUESTED** [review]
travis did not run for this PR:
> GitHub payload is missing a merge commit (mergeable_state: "unknown", merged: false)

This usually happens if PRs are not mergeable due to conflicts. Let's see if the build will run after the changes.

**Comment by eddycharly** [comment]
Github was down yesterday evening, that's probably why travis was not updated correctly.

File: Middleware/GuzzleRequestSpanning.php (hunk 1 above)
Comment (eddycharly) [reply Middleware/GuzzleRequestSpanning.php]: done

File: Middleware/GuzzleRequestSpanning.php [hunk 2]
```diff
@@ -30,11 +32,39 @@ public function __invoke(callable $handler): callable
...
+    private function onFulfilled(ResponseInterface $response): ResponseInterface
+    {
+        $this->requestSpanning->finish($response->getStatusCode());
+        $this->tracing->finishActiveSpan();
+        return $response;
+    }
+
+    private function onRejected(RequestException $exception): void
+    {
+        // TODO: remove this if $this->requestSpanning->finish adds the error tag (to avoid duplicate)
```
Comment (cawolf) [inline Middleware/GuzzleRequestSpanning.php]: This TODO is obsolete: `finish()` does not add the error tag. Even if it did: tags are unique and setting tags again overwrite the previous value.

File: Middleware/GuzzleRequestSpanning.php [hunk 3]
```diff
@@ -30,11 +32,39 @@ public function __invoke(callable $handler): callable
             $promise = $handler($request, $options);
             return $promise->then(
                 function (ResponseInterface $response) {
-                    $this->requestSpanning->finish($response->getStatusCode());
-                    $this->tracing->finishActiveSpan();
-                    return $response;
+                    return $this->onFulfilled($response);
```
Comment (cawolf) [inline Middleware/GuzzleRequestSpanning.php]: Just an idea: too keep the focus within this layer, we could also make `onFulfilled()` a `void` function (like `onRejected()`), and `return $response;`.

This way, the developer does not need to dive into `onFulfilled()` to grasp, what is returned from this function.

**Review by cawolf — CHANGES_REQUESTED** [review]

File: Middleware/GuzzleRequestSpanning.php (hunk 2 above)
Comment (eddycharly) [reply Middleware/GuzzleRequestSpanning.php]: I will remove the todo as it’s now part of the finish. 
There’s no guarantee that tags are unique.
For exemple in go tags are not and jaeger query shows a warning.

Comment (cawolf) [reply Middleware/GuzzleRequestSpanning.php]: Ah, I didn't know about these differences in language specific implementations. I just saw this in the PHP specification:
> If there is a pre-existing tag set for key, it is overwritten.

Comment (eddycharly) [reply Middleware/GuzzleRequestSpanning.php]: I thought the same at first but it looks like it's implementation dependent.
It's the case with jaeger-php though.

File: Middleware/GuzzleRequestSpanning.php (hunk 3 above)
Comment (eddycharly) [reply Middleware/GuzzleRequestSpanning.php]: done

**Review by cawolf — APPROVED** [review]
LGTM!

**Pull request merged**